examples/albert/run_first_peer.py

```diff
@@ -1,45 +1,159 @@
+            throughput=coordinator_args_dict.pop('bandwidth'),
+            target_batch_size=adjusted_target_batch_size, client_mode=coordinator_args_dict.pop('client_mode'),
+            verbose=True, start=True, **coordinator_args_dict
+        )
+        self.previous_timestamp = time.time()
+
+    def is_time_to_save_state(self, cur_step):
+        if self.save_checkpoint_step_interval is None:
+            return False
+        elif cur_step - self.previous_step >= self.save_checkpoint_step_interval:
+            return True
+        else:
+            return False
+
+    def save_state(self, cur_step):
+        self.collaborative_optimizer.load_state_from_peers()
+        self.previous_step = cur_step
+
+    def is_time_to_upload(self):
+        if self.repo_path is None:
+            return False
+        elif time.time() - self.previous_timestamp >= self.upload_interval:
+            return True
+        else:
+            return False
+
+    def upload_checkpoint(self, current_loss):
+        self.model.save_pretrained(self.repo_path)
+        torch.save(self.collaborative_optimizer.opt.state_dict(), f"{self.repo_path}/optimizer_state.pt")
+        self.previous_timestamp = time.time()
+        try:
+            subprocess.run("git add --all", shell=True, check=True, cwd=self.repo_path)
+            current_step = self.collaborative_optimizer.collaboration_state.optimizer_step
+            subprocess.run(f"git commit -m 'Step {current_step}, loss {round(current_loss, 3)}'", shell=True, check=True, cwd=self.repo_path)
+            subprocess.run("git push", shell=True, check=True, cwd=self.repo_path)
+        except subprocess.CalledProcessError as e:
+            logger.warning("Error while uploading model:", e.output)
+
+
 if __name__ == '__main__':
-    parser = argparse.ArgumentParser()
-
-    parser.add_argument('--address', type=str, required=False,
-                        help="this machine's network address. Use public IP for global experiments, "
-                             "local address for private runs.")
-    parser.add_argument('--listen_on', type=str, default='0.0.0.0:*', required=False,
-                        help="'localhost' for local connections only, '0.0.0.0' for ipv4 '[::]' for ipv6")
-    parser.add_argument('--refresh_period', type=float, default=30, required=False,
-                        help="coordinator will fetch keys from DHT once in this many seconds")
-    parser.add_argument('--experiment_prefix', type=str, required=True,
-                        help="a prefix where peers store their metrics for aggregation")
-    parser.add_argument('--wandb_project', type=str, required=True,
-                        help="Weights & Biases project name to publish learning curves")
-
-    args = parser.parse_args()
-    if args.address is None:
+    parser = HfArgumentParser(CoordinatorArguments)
+    coordinator_args = parser.parse_args_into_dataclasses()[0]
+    coordinator_args_dict = asdict(coordinator_args)
```

Comment:
When I was suggesting that we separate the arguments into groups, I also meant that we can parse them separately into several groups of arguments (see examples in the Transformers library or even in the run_trainer script)

This way, you 
1) won't have to repeatedly call pop() for the arguments, because they won't be passed into wrong classes
2) ideally, won't have to use asdict() unless the very moment when you're initializing internal Hivemind classes